src/coreclr/nativeaot/Runtime/eventpipe/CMakeLists.txt

```diff
@@ -151,8 +182,11 @@ if(CLR_CMAKE_HOST_UNIX)
     set_source_files_properties(${EVENTPIPE_SOURCES} PROPERTIES COMPILE_OPTIONS -xc++)
 endif(CLR_CMAKE_HOST_UNIX)
 
-if (WIN32)
+if (CLR_CMAKE_TARGET_WIN32)
     set_source_files_properties(${EVENTPIPE_SOURCES} PROPERTIES COMPILE_FLAGS "/FI\"${RUNTIME_DIR}/eventpipe/NativeaotEventPipeSupport.h\"")
+    set_source_files_properties(${GEN_EVENTPIPE_PROVIDER_SOURCES} PROPERTIES COMPILE_FLAGS "/FI\"${RUNTIME_DIR}/eventpipe/NativeaotEventPipeSupport.h\"")
+    set_source_files_properties(${GEN_EVENTPIPE_PLAT_AGNOSTIC_SOURCES} PROPERTIES COMPILE_FLAGS "/FI\"${RUNTIME_DIR}/eventpipe/NativeaotEventPipeSupport.h\"")
```

Comment:
Maybe just move this all down to group with `AOT_EVENTTRACE_SOURCES` on line 229?